docs/troublehooting.md

```diff
@@ -0,0 +1,242 @@
+1) if there are any messages in GCk6 UI
+2) if there are any messages in the output of `k6 cloud` command
+3) the resources and their logs, the same way as with [standalone `TestRun` deployment](#testrun-deployment)
+
+## Common scenarios
+
+### Where are my env vars...
+
+Some tricky cases with environment variables are described in [this doc](./env-vars.md).
+
+### Tags are not working?!
+
+Currently, tags are a rather common source of frustration in usage of k6-operator. For example:
+
+```yaml
+  arguments: --tag product_id="Test A"
+  # or
+  arguments: --tag foo=\"bar\"
+```
+
+Passing the above leads to parsing errors which can be seen in the logs of either initializer or runner Pod, e.g.:
+```bash
+time="2024-01-11T11:11:27Z" level=error msg="invalid argument \"product_id=\\\"Test\" for \"--tag\" flag: parse error on line 1, column 12: bare \" in non-quoted-field"
+```
+
+This is a standard problem with escaping the characters, and there's even an [issue](https://github.com/grafana/k6-operator/issues/211) that can be upvoted.
+
+### Initializer logs an error but it's not about tags
+
+Often, this happens because of lack of attention to the [highly recommended](#⚠️-highly-recommended-⚠️) step. One more command that can be tried here is to run the following:
+
+```bash
+k6 inspect --execution-requirements script.js
+```
+
+This command is a shortened version of what initializer Pod is executing. If the above command produces an error, it is definitely a problem with the script and should be first solved outside of k6-operator. The error itself may contain a hint to what is wrong, for instance a syntax error.
+
+If standalone `k6 inspect --execution-requirements` executes successfully, then it's likely a problem with `TestRun` deployment specific to your Kubernetes setup. Recommendations here:
+- read carefully the output in initializer Pod: is it logged by k6 process or by something else?
+  - :information_source: k6-operator expects initializer logs to contain only the output of `k6 inspect`. If there's any other log line present, then k6-operator will fail to parse it and the test will not start. ([issue](https://github.com/grafana/k6-operator/issues/193))
+- check events in initializer Job and Pod as they may contain another hint about what is wrong
+
+### Non-existent ServiceAccount
+
+ServiceAccount can be defined as `serviceAccountName` and `runner.serviceAccountName` in PrivateLoadZone and TestRun CRD respectfully. If the specified ServiceAccount does not exist, k6-operator will successfully create Jobs but corresponding Pods will fail to be deployed, and k6-operator will wait indefinitely for Pods to be `Ready`. This error can be best seen in the events of the Job:
+
+```bash
+kubectl describe job plz-test-154546-1
+...
+Events:
+  Warning  FailedCreate  57s (x4 over 2m7s)  job-controller  Error creating: pods "plz-test-154546-1-" is forbidden: error looking up service account plz-ns/plz-sa: serviceaccount "plz-sa" not found
```

Comment:
```suggestion
  Warning  FailedCreate  57s (x4 over 2m7s)  job-controller  Error creating: pods "plz-test-xxxxxx-x-" is forbidden: error looking up service account plz-ns/plz-sa: serviceaccount "plz-sa" not found
```
Also...not sure if the trailing "-" was originally a typo, or valid. 🤔

Reply:
Yes, actually. It's a funny thing but apparently job-controller outputs the prefix for future pod here, hence with hyphen.